### PR TITLE
WIP: add support for byt5 and mt5

### LINKED ISSUE #1655: Huggingface demos use an older version (4.6.1) of transformers, which has a bug in the implementation of t5
The Huggingface/demos requirements file [specifies](https://github.com/NVIDIA/TensorRT/blob/main/demo/HuggingFace/requirements.txt#L17) `transformers==4.6.1`

It is my understanding from [this PR in `transformers`](https://github.com/huggingface/transformers/pull/11621) that in 4.6.1 and before, T5's decoder used the encoder's attention mask when decoding and that the changes introduced in that PR (which show up in all versions of `transformers>4.6.1`) change the signature of T5 so that the decoder can have its own attention masking.

(Note: I didn't use the normal issue template because this isn't really a bug in TRT, but it seems worth being aware of.)

### PR DESCRIPTION
**Pull request by sam-writer**
closes #1655

Hi, sorry to open a WIP PR, but I needed to run byt5 in TRT, which meant supporting newer versions of `transformers`. Most of the changes are in the notebook, and I am not sure if the style is in line with your vision. My goal was to support all versions of t5, including mt5 and byt5 and community-made fine-tuned checkpoints. That made it impossible to pre-register all the model names and sizes.

Happy to make any requested modifications.

### CODE REVIEW

**Comment by sam-writer** [comment]
In the section `Full model inference benchmark`, if I set `use_cuda=True` I sometimes get errors like:

>[2021-12-22 21:57:39,889][OSS][WARNING] Unable to execute program using cuda compatible device: The expanded size of the tensor (63) must match the existing size (64) at non-singleton dimension 0.  Target sizes: [63].  Tensor sizes: [64]
> [2021-12-22 21:57:39,890][OSS][WARNING] Retrying using CPU only.

I am unclear if those were already present, since `use_cuda` was `False`.

**Comment by sam-writer** [comment]
Also, AFAICT, the TRT implementation of T5's decoder doesn't use `past_key_values`, which [according to the HuggingFace docs](https://huggingface.co/docs/transformers/model_doc/t5#transformers.T5Model.forward.past_key_values) speed up decoding.

I assume this is because it would make the decoder variadic, and like ORT, TRT does not like this. If that is the case, could we do something like [fastT5](https://github.com/Ki6an/fastT5/blob/master/fastT5/onnx_models.py#L57) and have 2 decoders: one for the first step, which computes `past_key_values` but does not use them, and then a decoder for all subsequent steps, which takes and returns `past_key_values`?

**Comment by sam-writer** [comment]
@ttyio are you a good person to ping about this?

**Comment by ttyio** [comment]
> @ttyio are you a good person to ping about this?

Thanks for the contribution. but sorry I am not familiar with this sample. Adding @rajeevsrao to review, thanks!
